ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs

```diff
@@ -490,11 +490,22 @@ extendImportViaParent df parent child (L l it@ImportDecl{..})
       -- we need change the ann key from `[]` to `:` to keep parens and other anns.
       unless hasSibling $
         transferAnn (L l' $ reverse pre) (L l' [x]) id
+
+      let lies' = reverse pre ++ [x]
 #else
-          x :: LIE GhcPs = reLocA $ L l'' $ IEThingWith listAnn parentLIE NoIEWildcard [childLIE]
           listAnn = epAnn srcParent [AddEpAnn AnnOpenP (epl 1), AddEpAnn AnnCloseP (epl 0)]
+          x :: LIE GhcPs = reLocA $ L l'' $ IEThingWith listAnn parentLIE NoIEWildcard [childLIE]
+
+      x <- pure $ setEntryDP x (SameLine $ if (not (null pre)) then 1 else 0)
+
+      let
+
+          fixLast = if not (null pre) then first addComma else id
```

Comment:
```suggestion
      let hasSibling = not (null pre)
      x <- pure $ setEntryDP x (SameLine $ if hasSibling then 1 else 0)

      let

          fixLast = if hasSibling then first addComma else id
```

Reply:
There is an opportunity to factor some logic out in the same module:

https://github.com/haskell/haskell-language-server/blob/7af4e587fe5696524821390dd381bd024a8ca14b/ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs#L378-L381